### PR TITLE
Add reusable Framework features from lighthouse

### PR DESCRIPTION
**Pull request by tpantelis**
- Framework.ExecWithOptions function
- Service-related functions
- Deployment-related functions

### CODE REVIEW

File: test/e2e/framework/deployments.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 
 	. "github.com/onsi/gomega"
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (mkolesnik) [inline test/e2e/framework/deployments.go]: This makes no sense :)

File: test/e2e/framework/services.go [hunk 2]
```diff
@@ -3,6 +3,8 @@ package framework
 import (
 	"fmt"
 
+	"github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
```
Comment (mkolesnik) [inline test/e2e/framework/services.go]: Here as well

**Comment by tpantelis** [comment]
> There's some double imports of core/v1, sometimes as v1 and sometimes as corev1, and I think we need to consistently name it corev1 (since there's also apiv1 imports)

yeah that was unintentional resulting from the copy - thanks for noticing it.

File: test/e2e/framework/deployments.go (hunk 1 above)
Comment (tpantelis) [reply test/e2e/framework/deployments.go]: done

File: test/e2e/framework/services.go (hunk 2 above)
Comment (tpantelis) [reply test/e2e/framework/services.go]: done

**Pull request merged**